File: EventApp.API/Models/Event.cs
using System;
using System.Collections.Generic;

namespace EventApp.API.Models
{
    public class Event
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Headline { get; set; }
        public string Description { get; set; }
        public int VenueId { get; set; }
        public Venue Venue { get; set; }
        public DateTime Date { get; set; }
        public DateTime Created { get; set; }
        public string PhotoURL { get; set; }
        public bool Approved { get; set; }
        public ICollection<Image> Images { get; set; }
        public User User { get; set; }
        public int UserId { get; set; }
        public Subcategory Subcategory { get; set; }
        public int SubcategoryId { get; set; }
        public ICollection<TicketCategory> TicketCategories { get; set; }
    }
}

File: EventApp-SPA/src/app/models/event.ts
import { TicketCategory } from './ticketCategory';
import { User } from './user';
import { Image } from './image';
import { Venue } from './venue';
import { Subcategory } from './subcategory';

export interface Event {
    id: number;
    name: string;
    headline: string;
    description: string;
    venue: Venue;
    date: Date;
    created: Date;
    photoURL?: string;
    finished: boolean;
    approved: boolean;
    rejected: boolean;
    images?: Image[];
    user: User;
    subcategory: Subcategory;
    ticketCategories?: TicketCategory[];
}
